bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/ScriptEngineManagerImpl.java

```diff
@@ -163,6 +169,22 @@ public void loadScript(String engineIdentifier, InputStreamReader scriptData) {
         }
     }
 
+    private void tryExtractAndInjectFilename(String engineIdentifier, ScriptEngine engine) {
+        try {
+            if(engine.getContext().getAttribute(ScriptEngine.FILENAME) == null) {
+                URI fileURI = new URL(engineIdentifier).toURI();
+                if ("file".equals(fileURI.getScheme())) {
+                    File maybeScriptFile = Paths.get(fileURI).toFile();
+                    if (maybeScriptFile.isFile()) {
+                        engine.getContext().setAttribute(ScriptEngine.FILENAME, maybeScriptFile.getName(), ScriptContext.ENGINE_SCOPE);
+                    }
```

Comment:
Maybe get some logging in there in case engineIdentifier is not a file?
```suggestion
                    } else {
                        logger.debug("engineIdentifier '{}' is not a file", engineIdentifier);
                    }
```